x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/PositionToXContent.java

```diff
@@ -108,6 +109,10 @@ protected XContentBuilder valueToXContent(XContentBuilder builder, ToXContent.Pa
                         // https://github.com/FasterXML/jackson-dataformats-binary/issues/366
                         val = BytesRef.deepCopyOf(scratch);
                     }
+
+                    if (columnInfo.name().equalsIgnoreCase(MetadataAttribute.TSID_FIELD)) {
```

Comment:
I don't think we should add this workaround. We already have the _tsid type. Can you change the encoder for tsid to `TopNEncoder.DEFAULT_UNSORTABLE` cases in LocalExecutionPlanner#planTopN?